COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,671 @@
-        if self.singleHist:
-            self.PlotDict["singleHistogram"] = self.singleHist
-            self.PlotDict["All"] = self.PlotDict["All"] + self.singleHist
-
-        # Reconfig the plots to be sure
-        self.PlotDict["All"] = config_layout(self.PlotDict["All"], **self.config[self.name].get("Layout", {}))
-
-        return self.PlotDict
\ No newline at end of file
+        self.donts_mos = ["timestamp", "voltage", "Voltage", "Stepsize", "Wait", "Stepsize", "Frequency", "x", "N", "Current"]  # don't plot these.
+        self.basePlots5 = plot_all_measurements(self.data, self.config, self.xaxis, self.name, do_not_plot=self.donts_mos, keys=cv_files)
+        self.PlotDict["BasePlots_MOS"] = self.basePlots5
+
+        # Add flat bandage voltage point to the Capacity curve
+        if self.config["IV_PQC"].get("CapacityCopy", {}).get("findFlatBandVoltage", False):
+            try:
+                if self.basePlots5.Overlay.MOS_CV_CURVES.children:
+                    clone_plot = self.basePlots5.Overlay.MOS_CV_CURVES.opts(clone=True)
+                else:
+                    clone_plot = self.basePlots5.Curve.MOS_CV_CURVES.opts(clone=True)
+                fBestimation = self.find_flatBand_voltage(clone_plot, self.data, self.config, indexMax, indexMin, df, cv_files, voltage_value_of_max_firstder, PlotLabel="Flat band voltage estimation")
+            except Exception as err:
+                self.log.warning("No flat band voltage calculation possible... Error: {}".format(err))
+
+        # Do these plots for the analysis of one single cv mos file
+        if len(cv_files) == 1:
+            self.PlotDict["BasePlots_MOS"] += fBestimation[0]
+            self.PlotDict["BasePlots_MOS"] += derivative_interpolation_plot
+            self.PlotDict["BasePlots_MOS"] += secondderivative_interp_plot
+            self.PlotDict["BasePlots_MOS"] += capa_interp_plot
+            self.PlotDict["BasePlots_MOS"] += capa_interp_plot * max_firstder_plot * derivative_interpolation_plot * secondderivative_interp_plot
+            self.PlotDict["BasePlots_MOS"] += fBestimation[6] * max_firstder_plot * derivative_interpolation_plot
+
+        # Add a Table that shows the differents analysis parameters values
+        df2 = pd.DataFrame({"Name": cv_files, "Flatband Voltage second_derivative (V)": fbvoltage,
+             'Flatband Voltage first_derivative (V)': fbvoltage_firstderivative,
+             'Accumulation capacitance (F)': Accum_capacitance_list,
+             'Accumulation capacitance normalized (F/cm^2)': Accum_capacitance_normalized_list, 'Tox (nm)': Tox_list,
+             'Nox (cm^-2)': Nox_list})
+        table1 = hv.Table(df2, label='Mos analysis')
+        table1.opts(width=1300, height=800)
+        # Do plots
+        self.PlotDict["BasePlots_MOS"] += table1
+
+        return self.PlotDict["BasePlots_MOS"]
+
+    def analysis_diode(self, df):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Global variable 'fdestimation' undefined at the module level](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)